src/components/map/Map.tsx

```diff
@@ -126,21 +128,45 @@ export const Map = ({
             zIndex={1}
           />
         )}
-        {locations?.map((marker, index) => (
-          <Marker
-            centerOffset={marker.iconAnchor || { x: 0, y: -(MARKER_ICON_SIZE / 2) }}
-            coordinate={marker.position}
-            identifier={marker.id}
-            key={`${index}-${marker.id}`}
-            onPress={() => onMarkerPress?.(marker.id)}
-            zIndex={selectedMarker && marker.id === selectedMarker ? 1010 : 1}
-          >
-            <MapIcon
-              iconColor={selectedMarker && marker.id === selectedMarker ? colors.accent : undefined}
-              iconName={marker.iconName}
-            />
-          </Marker>
-        ))}
+        {locations?.map((marker, index) => {
+          const isActiveMarker = selectedMarker && marker.id === selectedMarker;
+
+          return (
+            <Marker
+              centerOffset={marker.iconAnchor || { x: 0, y: -(MARKER_ICON_SIZE / 2) }}
+              coordinate={marker.position}
+              identifier={marker.id}
+              key={`${index}-${marker.id}`}
+              onPress={() => onMarkerPress?.(marker.id)}
+              zIndex={isActiveMarker ? 1010 : 1}
+            >
+              {!!marker.iconName && marker.iconName != 'ownLocation' ? (
+                <>
+                  <MapIcon iconColor={isActiveMarker ? colors.accent : undefined} />
+                  <View
+                    style={[
+                      styles.mapIconOnLocationMarker,
+                      isActiveMarker ? styles.mapIconOnLocationMarkerActive : undefined
+                    ]}
+                  >
+                    <MapIcon
+                      iconColor={colors.surface}
+                      iconName={marker.iconName}
+                      iconSize={MARKER_ICON_SIZE / 3.25}
+                    />
+                  </View>
+                </>
+              ) : (
+                <MapIcon
+                  iconColor={
+                    selectedMarker && marker.id === selectedMarker ? colors.accent : undefined
+                  }
+                  iconName={marker.iconName}
```

Comment:
We check `marker.iconName` above and check whether there is a value. This is the section where `marker.iconName` has no value. Should we add the `iconName` prop here?

Reply:
here? there is `iconName={marker.iconName}`, which can be undefined to use `'location'`.

Reply:
should be done with https://github.com/smart-village-solutions/smart-village-app-app/pull/926#discussion_r1335790276